aws-sdk-core/lib/aws-sdk-core/errors.rb

```diff
@@ -53,6 +53,52 @@ class MissingCredentialsError < RuntimeError; end
     # Raised when a {Service} is constructed and region is not specified.
     class MissingRegionError < ArgumentError; end
 
+    # Raised when attempting to connect to an endpoint and a `SocketError`
+    # is received from the HTTP client. This error is typically the result
+    # of configuring an invalid `:region`.
+    class NoSuchEndpointError < RuntimeError
+
+      def initialize(options = {})
+        @endpoint = options[:endpoint]
+        @original_error = options[:original_error]
+        super(<<-MSG)
+Encountered a `SocketError` while attempting to connect to:
+
+  #{endpoint.to_s}
+
+This is typically the result of an invalid `:region` option.
```

Comment:
Could be worth mentioning poorly formatted custom endpoints here, which is also a common cause.